extensions/2.0/Khronos/KHR_behavior/README.md

```diff
@@ -62,6 +62,290 @@ From these requirements, the main features can be deducted:
+
+### Actions categorized
+
+Only Level 0 actions for now, probably not a complete list:
+
+`active`:
+- start
+- select
+
+`passive`:
+- hoverStart
+- hoverEnd
+- proximityStart
+- proximityEnd
+- inViewStart
+- inViewEnd
+
+`stateless`:
+- interpTo
+- waitAll
+- gate
+- delay
+- playSound
+- terminate
+
+`stateful`:
+- animation
+- moveTo
+- turnTo
+- visibility
+- flipflop
+- multiGate
+- doOnce
+
+Most of these are described below in the original document. Only `flipflop` and `multiGate` have multiple outputs. The `terminate` action lists another action in its parameter list; it will stop that action and any others downstream from it when triggered, effectively canceling an entire behavior and leaving the scene at its current state.
+
+## Function object
+
+*While `action` objects represent the async `behaviorNodes`, `function` objects represent the synchronous or immediate `behaviorNodes`, focusing on values rather than events.*
+
+In addition to the `actions` list, there is also a `functions` list. Each `function` takes one or more `value` parameters as input and produces one `value` output. Any `parameter` in either an `action` or a `function` can be either a literal constant or a reference to a `function` output.
+
+Each `function` is synchronous and a given `function` graph can be thought of as a call stack, beginning with an `action` parameter and calling each parameter `function` in turn up the stack until it terminates at either a constant or a query. The stacks are then unwound down to produce the final requested parameter `value`. 
+
+Execution is deterministic because the `functions` are all synchronous and pure, meaning they have no state and no side-effects. They cannot affect the scene states they are querying directly: only `action` nodes can do that. A `function` graph runs before the `action` that calls it via its parameter list, and after that `action`'s input `action` completed. This means if both of these actions are changing the same scene state value, that value will be deterministically in the final state of the input `action` when queried. 
+
+In addition to querying scene properties and animation states, the state of `passive` triggers can also be queried, which can be especially useful in combination with a gate `action`. 
+
+### Security requirements for Level 0
+
+Replacing constant parameters with `functions` is allowed in general across Level 0, as long as all functions are pure, stateless, and side-effect-free. Like `actions`, `functions` must only reference indices less than their own to prevent recursion. They do not impact the fundamental restriction that the Turing machine can only advance via explicit user activation.
```

Comment:
I believe this means Level 0 can incorporate most of what we've been describing for Level 1: it comes down to just a restriction on recursion (by index) and a requirement for user activation.